src/fastq/call_consensus_reads.rs

```diff
@@ -173,170 +263,252 @@ pub fn call_consensus_reads_from_paths(
     eprintln!("Writing output to:\n    {}\n    {}", fq1_out, fq2_out);
 
     match (fq1.ends_with(".gz"), fq2.ends_with(".gz"), fq1_out.ends_with(".gz"), fq2_out.ends_with(".gz")) {
-        (false, false, false, false) => call_consensus_reads(
-            &mut fastq::Reader::from_file(fq1)?,
-            &mut fastq::Reader::from_file(fq2)?,
-            &mut fastq::Writer::to_file(fq1_out)?,
-            &mut fastq::Writer::to_file(fq2_out)?,
+        (false, false, false, false) => CallNonOverlappingConsensusRead::new(
+            &mut fastq::Reader::from_file(fq1).expect("Couldn't open first input file"),
+            &mut fastq::Reader::from_file(fq2).expect("Couldn't open second input file"),
+            &mut fastq::Writer::to_file(fq1_out).expect("Couldn't open first output file"),
+            &mut fastq::Writer::to_file(fq2_out).expect("Couldn't open second output file"),
```

Comment:
Good idea to have specific error messages here. Instead of panicing, we should finally start using the [failure crate](https://boats.gitlab.io/failure/), which allows to add custom messages to errors on the fly.

Reply:
I just added a very rough sketch of how this could work.

We would need to implement custom error types to really leverage the `failure` crate. I am not sure yet, if implementing them per tool or once for the whole crate is the right way to handle this.

Either way, we would need rework all functions to replace `Box<dyn Error>` with `Context<StillToBeNamedErrorType>` (where I would be tempted to implement `StillToBeNamedErrorType` as [an Enum](https://github.com/rust-lang-nursery/failure/blob/master/book/src/derive-fail.md#enums)).

It might also be helpful to modify the Error types returned by rust-bio. They do not implement `Sync` and `Send`, which is required to easily repackage them with `failure`.

Any thoughts on that?

Reply:
You are right, this is quite some work to do, and indeed has to start with rust-bio. Let us first merge this and focus on applying minimap for getting the starcode part faster. If anybody wants to have a look into migrating rust-bio to failure you can of course do that in parallel.